weasyl/search.py

```diff
@@ -318,22 +342,22 @@ def get_ids(names):
 
     if backid:
```

Comment:
I thoroughly hate this code. I understand the awkwardness in restructuring it into a if backid: elif nextid: else: tree because of the shared behavior around next_count. But there's too much magic here to understand for someone who didn't write it imo.

For example, why is ret reversed in one case? At over 200 lines of Python SQL logic they'll have long since lost track. At minimum it should have comments.